ge0/geo_url_parser.hpp

```diff
@@ -23,13 +23,15 @@ class GeoURLInfo
   double m_lat;
   double m_lon;
   double m_zoom;
+  std::string m_query;
+  std::string m_label;
 };
 
 class DoubleGISParser
 {
 public:
   DoubleGISParser();
-  bool Parse(url::Url const & url, GeoURLInfo & info) const;
+  [[nodiscard]] bool Parse(url::Url const & url, GeoURLInfo & info) const;
```

Comment:
We agreed some time ago that tags like [[nodiscard]] in C++ are _syntax garbage_ No real value ..

Reply:
I don't know who agreed and why, but I can use old school __attribute__((unused_result)) instead. The value is that parsing result will not be ignored.

Reply:
@vng it may be ok in this case.

Reply:
Come on, just remove it. We have thousands of bool functions already. Will put it everywhere?

Reply:
No, you need to put exceptions everywhere to be fully C++ compliant ;) [[nodiscard]]  is not bad for C-style error handling because it adds compile-time checks. However, it wasn't the purpose of this patch and I plan to remove it from the PR.

Reply:
Removed.